contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -0,0 +1,114 @@
+with Ada.Directories; use Ada.Directories;
+with Ada.Exceptions;
+with Ada.Strings.Unbounded; use Ada.Strings.Unbounded;
+with Ada.Text_IO; use Ada.Text_IO;
+
+with GNAT.Traceback.Symbolic;
+with GNATCOLL.Opt_Parse; use GNATCOLL.Opt_Parse;
+with GNATCOLL.Traces;
+
+with Langkit_Support.Diagnostics; use Langkit_Support.Diagnostics;
+with Langkit_Support.Diagnostics.Output;
+use Langkit_Support.Diagnostics.Output;
+with Langkit_Support.Text;        use Langkit_Support.Text;
+
+with Liblktlang.Analysis;
+with Liblktlang.Common;
+
+procedure Lkt_Toolbox is
+
+   package Arg is
+
+      Parser : Argument_Parser := Create_Argument_Parser
+        (Help => "Lkt toolbox. Toolbox like command line frontend for the "
+                 & "LKT langkit library.");
+
+      package Files is new Parse_Positional_Arg_List
+        (Parser   => Parser,
+         Name     => "files",
+         Arg_Type => Unbounded_String,
+         Help     => "The files to parse");
+
+      package Check_Only is new Parse_Flag
+        (Parser => Parser,
+         Short  => "-C",
+         Long   => "--check-only",
+         Help   => "Only output the errors");
+
+   end Arg;
+
+   use Liblktlang;
+
+   procedure Print_Semantic_Result
+     (S : Analysis.Semantic_Result; Buffer : in out Text_Buffer);
+   --  Print a semantic result
+
+   ---------------------------
+   -- Print_Semantic_Result --
+   ---------------------------
+
+   procedure Print_Semantic_Result
+     (S : Analysis.Semantic_Result; Buffer : in out Text_Buffer)
+   is
+   begin
+      if Analysis.Error_Message (S) /= "" then
+         declare
+            Diag : constant Diagnostic :=
+              (Analysis.Node (S).Sloc_Range,
+               To_Unbounded_Text (Analysis.Error_Message (S)));
+         begin
+            Print_Diagnostic
+              (Diag, Buffer,
+               Simple_Name (Analysis.Node (S).Unit.Get_Filename));
+         end;
+      elsif
+        not Arg.Check_Only.Get
+        and then not Analysis.Result_Type (S).Is_Null
+      then
+         Put_Line ("Expr " & Analysis.Node (S).Image);
+         Put_Line ("     has type " & Analysis.Result_Type (S).Image);
+         New_Line;
+      elsif
+      not Arg.Check_Only.Get
+        and then not Analysis.Result_Ref (S).Is_Null
+      then
+         Put_Line ("Id   " & Analysis.Node (S).Image);
+         Put_Line ("     references " & Analysis.Result_Ref (S).Image);
+         New_Line;
+      end if;
+   end Print_Semantic_Result;
+
+   Ctx : constant Analysis.Analysis_Context := Analysis.Create_Context;
+begin
+   GNATCOLL.Traces.Parse_Config_File;
+
+   if Arg.Parser.Parse then
+      for File_Name of Arg.Files.Get loop
+         declare
+            Unit : constant Analysis.Analysis_Unit
+              := Ctx.Get_From_File (To_String (File_Name));
+         begin
+            if Unit.Diagnostics'Length > 0 then
+               for Diagnostic of Unit.Diagnostics loop
+                  Put_Line (To_Pretty_String (Diagnostic));
```

Comment:
Just curious: why did not you use `Print_Diagnostic` here?

Reply:
Good catch, that's actually a remain of the old ada test main that I did before I had `Print_Diagnostics`.

That said, I'm not sure syntax errors are going to be greatly useful with `Print_Diagnostics` because their sloc is not generally very helpful, but let's use it for consistency.